File: app/src/main/java/com/attendance/data/source/remote/AttendResultsRemoteDataSource.java
package com.attendance.data.source.remote;

import com.attendance.data.AttendResult;

import java.util.ArrayList;
import java.util.List;

import rx.Observable;
import rx.Subscriber;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

/**
 * Created by peiqin on 3/5/2017.
 */

public class AttendResultsRemoteDataSource {

    private AttendResultsRemoteDataSource() {
    }

    private static class SingletonHolder {
        private static final AttendResultsRemoteDataSource INSTANCE = new AttendResultsRemoteDataSource();
    }

    public static AttendResultsRemoteDataSource getInstance() {
        return SingletonHolder.INSTANCE;
    }

    public void getAttendResultList(Subscriber<AttendResult> subscriber) {

        List<AttendResult> list = new ArrayList<>();
        for (int i = 0; i < 5; i++) {
            list.add(new AttendResult("student" + "", i + "", i + "", i + "", i + ""));
        }

        Observable observable = Observable.from(list);
        observable.subscribeOn(Schedulers.io())
                .unsubscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(subscriber);

    }

}

File: app/src/main/kotlin/com.attendance/data/source/remote/AttendResultsRemoteDataSource.kt
package com.attendance.data.source.remote

import com.attendance.data.AttendResult

import java.util.ArrayList

import rx.Observable
import rx.Subscriber
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

/**
 * Created by peiqin on 3/5/2017.
 */

class AttendResultsRemoteDataSource private constructor() {

    private object SingletonHolder {
        val INSTANCE = AttendResultsRemoteDataSource()
    }

    fun getAttendResultList(subscriber: Subscriber<AttendResult>) {

        val list = ArrayList<AttendResult>()
        for (i in 0..4) {
            list.add(AttendResult("student" + "", i.toString() + "", i.toString() + "", i.toString() + "", i.toString() + ""))
        }

        val observable = Observable.from(list)
        observable.subscribeOn(Schedulers.io())
                .unsubscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(subscriber)

    }

    companion object {
        val instance: AttendResultsRemoteDataSource
            get() = SingletonHolder.INSTANCE
    }

}